python/ray/dag/compiled_dag_node.py

```diff
@@ -2058,10 +2087,7 @@ def __del__(self):
         monitor = getattr(self, "_monitor", None)
         if monitor is not None:
             # Teardown asynchronously.
-            # NOTE(swang): Somehow, this can get called after the CoreWorker
-            # has already been destructed, so it is not safe to block in
-            # ray.get.
-            monitor.teardown(wait=False)
+            monitor.teardown(wait=True)
```

Comment:
Will this wait indefinitely or delay process shutdown?

Also update the comment above